### DISCUSSION TITLE
What exceptions does a client see when record operations fail?

### DISCUSSION THREAD

**Discussion by replicant1** [pedroSG94/RootEncoder#1488] [Q&A]
Folks,
I'm wondering what exceptions a client to the `RootEncoder` will see if the client calls `RtmpCamera1.startStreamAndRecord(streamUrl, filePath)` and the device's disk gets full while the recording is in progress? The `startStreamAndRecord` method is defined to throw an `IOException`, but what about write-related errors that occur later while the stream is in progress? Is a `Runtime` exception thrown that the client can catch, or is it swallowed by the `RootEncoder`?
Thanks in advance!

**Comment by pedroSG94** [comment]
Hello,

The write errors should be catch here:
https://github.com/pedroSG94/RootEncoder/blob/master/library/src/main/java/com/pedro/library/util/AndroidMuxerRecordController.java#L154
I think I should create a callback to allow user capture it

**Comment by replicant1** [comment]
Hi Pedro,
Thanks for your prompt response! Do you have any specific ideas about how you would like to see that implemented? For example, would the caught exception be passed to a new method in the` RecordController.Listener `interface, perhaps? I may be in a position to submit a PR.
Thanks!

**Comment by pedroSG94** [comment]
Hello,

Yes, that will works.
If possible I want add this new method as default to make it optional to implement and keep backward compatibility. Something like this:
```java
default void onError(Exception e) {}
```
Also, I need to test full disk case to make sure that the error is properly catch.

**Reply by pedroSG94** [reply]
Remember implement this callback in both: AacMuxerRecordController and AndroidMuxerRecordController

**Comment by pedroSG94** [comment]
Added here:
https://github.com/pedroSG94/RootEncoder/commit/5c6e67d274460335c16c80d20c6f3606678ab7a8
You can use this gradle:
```gradle
  implementation 'com.github.pedroSG94.RootEncoder:library:b3e3675511'
```

**Reply by replicant1** [reply]
Thanks!
